### ISSUE TITLE
dial tcp 192.168.65.3:10255: connect: connection refuse

### ISSUE THREAD

**Issue by amitm02** [elastic/helm-charts#456]
**Chart version:**
7.5.2

**Kubernetes version:**
1.14.8

**Kubernetes provider:** E.g. GKE (Google Kubernetes Engine)
docker-desktop & digitalocean

On default helm install I get:

`2020-01-22T08:01:33.150Z	INFO	module/wrapper.go:252	Error fetching data for metricset kubernetes.volume: error doing HTTP request to fetch 'volume' Metricset data: error making http request: Get http://docker-desktop:10255/stats/summary: dial tcp 192.168.65.3:10255: connect: connection refused`

**Comment by jmlrt** [comment]
Hi @amitm02, can you give us more details about your issue?

Mentioning the product you are using (Elasticsearch? Filebeat? ...), the exact command you are using for chart deployment and your config (`helm get <release>`) would be a good start.

Answering to **all questions** in the [bug report issue template](https://github.com/elastic/helm-charts/issues/new?template=bug_report.md) would help us to help you.

**Comment by jmymy** [comment]
Looks like its an error with metricbeat pulling data from the kube-state-metrics as that is what metricbeat uses to grab advanced metricsets. Make sure you have defined the correct endpoint for your kube-state-metrics, or make sure you have it deployed.

**Comment by babadofar** [comment]
kube-state-metrics is deployed together with metricbeat helm chart so you could actually maintain that this should work. https://github.com/elastic/helm-charts/blob/master/metricbeat/requirements.yaml

**Comment by masterkain** [comment]
I deploy on EKS and have the same issue using 7.6.1 charts.

```
 2020-03-17T02:15:33.048Z    INFO    module/wrapper.go:252    Error fetching data for metricset kubernetes.container: error doing HTTP request to fetch 'container' Metricset data: error making http request: Get http://ip-192-168-133-222.eu-west-1.compute.internal:10255/stats/summary: dial tcp 192.168.133.2
│ 2020-03-17T02:15:38.735Z    INFO    module/wrapper.go:252    Error fetching data for metricset kubernetes.system: error doing HTTP request to fetch 'system' Metricset data: error making http request: Get http://ip-192-168-133-222.eu-west-1.compute.internal:10255/stats/summary: dial tcp 192.168.133.222:102
│ 2020-03-17T02:15:39.538Z    INFO    module/wrapper.go:252    Error fetching data for metricset kubernetes.pod: error doing HTTP request to fetch 'pod' Metricset data: error making http request: Get http://ip-192-168-133-222.eu-west-1.compute.internal:10255/stats/summary: dial tcp 192.168.133.222:10255: co
│ 2020-03-17T02:15:41.972Z    INFO    module/wrapper.go:252    Error fetching data for metricset kubernetes.node: error doing HTTP request to fetch 'node' Metricset data: error making http request: Get http://ip-192-168-133-222.eu-west-1.compute.internal:10255/stats/summary: dial tcp 192.168.133.222:10255:
│ 2020-03-17T02:15:43.047Z    INFO    module/wrapper.go:252    Error fetching data for metricset kubernetes.container: error doing HTTP request to fetch 'container' Metricset data: error making http request: Get http://ip-192-168-133-222.eu-west-1.compute.internal:10255/stats/summary: dial tcp 192.168.133.2
│ 2020-03-17T02:15:43.054Z    INFO    module/wrapper.go:252    Error fetching data for metricset kubernetes.volume: error doing HTTP request to fetch 'volume' Metricset data: error making http request: Get http://ip-192-168-133-222.eu-west-1.compute.internal:10255/stats/summary: dial tcp 192.168.133.222:102
│ 2020-03-17T02:15:48.735Z    INFO    module/wrapper.go:252    Error fetching data for metricset kubernetes.system: error doing HTTP request to fetch 'system' Metricset data: error making http request: Get http://ip-192-168-133-222.eu-west-1.compute.internal:10255/stats/summary: dial tcp 192.168.133.222:102
│
```

**Comment by jmymy** [comment]
Can you confirm kube-state-metrics is running? and what the KUBE_STATE_METRICS_HOSTS env var is set to in the running metricbeat container?

**Comment by jmlrt** [comment]
@amitm02, @jmymy, @babadofar, @masterkain 

We just merged #471 which now use kubelet secure port (`10250`) instead of read-only port (`10255`) to fetch k8s metrics. We tested it on GKE and Docker-For-Mac without issues and think that should also solve issues on EKS.

Could you retry by installing Metricbeat chart [from master branch](https://github.com/elastic/helm-charts/blob/master/metricbeat/README.md#using-master-branch) and confirm if that fixes your issues?

**Comment by jmlrt** [comment]
This issue has no more activity.

I'm closing it, don't hesitate to reopen it if you have any question.
